### ISSUE TITLE
Add feedback capability to shot flow

### ISSUE THREAD

**Issue by johngruen** [mozilla-services/screenshots#4824]
@6a68 and i talked about implementing a very simple one-question survey flow in the add-on to help us better segment use cases.

TL;DR; I want to be able to deploy a survey to screenshot takers (regardless of what type of shot they're taking)  without having to make survey content ride the trains.

## User Flow

1. User takes a shot as normal.
2. They receive the notification indicating that their shot has been saved/copied/etc as usual.
3. After taking a shot, the `iframe` is not immediately torn down. Instead we show a modal with a single survey question.
4. The user can answer this question or click on the background to dismiss.
5. Either way, completing the survey tears down the `iframe` and completes the shot flow.
6. If the user elected to upload the shot, we pop the web ui in a new tab as usual.

![img_7375](https://user-images.githubusercontent.com/3323249/44912790-a9df9000-ad2b-11e8-8a87-3ffda4ae3483.jpg)

## Managing Surveys

The survey UI shell should be baked into the add-on so that from the add-on's perspective the flow will be something like this:

1. User starts Screenshots
2. Screenshots does a get on the survey server for the the latest survey
3. If there's a new survey, the add-on checks the the survey's filters to see if the user qualifies to take it.
4. If the user qualifies, the survey is displayed at the completion of the user's shot flow.
5. Once the user completes the survey; we post their result to a dashboard

Don't need to get too far in the weeds here, but I'm imagining the data we're sending around looks something like this: 

```
id, // Unique id of a survey
questionText, // Question text for survey
options: [], // Answer options for survey
filters: [] // Logic to filter exposure of survey
```

**Comment by 6a68** [comment]
Looks great. Do you think we'd want to include any additional data in the survey response, like whether the shot was a copy or download, whether the init was a right-click or a page action, or how many times the user had done screenshots stuff in that same session (Telemetry data)? This could be pluggable, like the filters, or it could be fixed per version and included in the POST data.

**Comment by ianb** [comment]
Some open questions for discussion:

- What server do we want to host surveys on?
- When is it OK to do a request to the server? E.g., are we okay doing at least one request for each use of Screenshots? Only as often as once a day?
- Are there any conditions we want to support?
- How do we want to handle localization? Just hardcode only en? Multiple survey endpoints for localized versions, if those exist?
- Do we want users to be able to tell us "never ask me about surveys"?
- Do we want to host the survey UI in-product, or on the web?
- Where do we want to send survey results?

**Comment by youwenliang** [comment]
Quick idea about the survey UI:
![survey](https://user-images.githubusercontent.com/8919308/45197106-21647080-b292-11e8-96c3-089c6376653d.png)
![survey thanks](https://user-images.githubusercontent.com/8919308/45197104-21647080-b292-11e8-91c7-d987787954b2.png)

**Comment by youwenliang** [comment]
I updated the design a bit:
https://mozilla.github.io/testpilot-assets/Firefox_Screenshots/FxA_Survey

Motion here:
![survey](https://user-images.githubusercontent.com/8919308/45545551-34a4ac80-b84d-11e8-94ee-d797c5adbf82.gif)

**Comment by johngruen** [comment]
@youwenliang looks great.

I think the ✅ can automatically close and the flow can continue by default after 2s

**Comment by johngruen** [comment]
> Looks great. Do you think we'd want to include any additional data in the survey response, like whether the shot was a copy or download, whether the init was a right-click or a page action, or how many times the user had done screenshots stuff in that same session (Telemetry data)? This could be pluggable, like the filters, or it could be fixed per Firefox version and included in the POST data.

Yes, i think it would be a good idea to pass in how the shot was taken, how it was initialized and the firefox version (for starters)

**Comment by 6a68** [comment]
Decisions from standup today, for initial version of the survey tool:

- Ship the survey and questions in Firefox 64
- No API endpoint
- Just show it once, to 10% of users
- Send the results to GA via sendEvent

**Comment by youwenliang** [comment]
> @youwenliang looks great.
> 
> I think the ✅ can automatically close and the flow can continue by default after 2s

Agree.

**Comment by johngruen** [comment]
@6a68 I think we're locking down a question. Tyler from the S&I team will review for us.

https://docs.google.com/document/d/1FCkh9mgoOd_z80FQ5Xqi_Y6TbzgeHsjoL44fOBAcFNk/edit#
